gui/directoryservice/views.py

```diff
@@ -426,8 +426,31 @@ def directoryservice_idmap_tdb2(request, id):
     })
 
 
+def directoryservice_idmap_script(request, id):
+    idmap_script = models.idmap_script.objects.get(id=id)
+
+    if request.method == "POST":
+        form = forms.idmap_script_Form(request.POST, instance=idmap_script)
+        if form.is_valid(): 
+            form.save()
+            return JsonResp(
+                request,
+                message="Idmap script successfully edited."
+            )
+        else:
+            return JsonResp(request, form=form)
+    else:
+        form = forms.idmap_script_Form(instance=idmap_script)
+
+    return render(request, 'directoryservice/idmap_script.html', {
+        'form': form
+    })
+
+
 def directoryservice_idmap_backend(request, obj_type, obj_id, idmap_type):
     data = utils.get_idmap(obj_type, obj_id, idmap_type)
+    if not data:
+        return
```

Comment:
In general you are right and that interconnectivity between js and py code drove me crazy :)

I try to solve here handling of the new, not coded yet backends, so they are shown properly, but we don't have code for them yet to put into the config. A warning is given instead.

So, `utils.get_idmap()` in this case returns `None`, as an object expected to be returned in case of success. So, AFAIU, we can't simply return [] or {} instead.

So, `None` is returned and `directoryservice_idmap_backend()` returns nothing in turn. It is a handler for the `/directoryservice/idmap_backend/*/*` URL, which we query from the dojo code within the `directoryservice_idmap_get_edit_url ()` function:

```javascript
    directoryservice_idmap_get_edit_url = function(eid, ds_type, ds_id) {
        var widget = registry.byId(eid);
        var idmap_backend = widget.get("value");
        var idmap_url = "/directoryservice/idmap_backend/" +
            ds_type + "/" + ds_id + "/" + idmap_backend + "/";
        var edit_url = null;
        var id = -1;

        //console.log("Idmap URL:", idmap_url);

        xhr.get(idmap_url, {
            sync: true,
            handleAs: 'json'
        }).then(function(data) {
                id = data.idmap_id;
            },
            function(error) {
                id = 0;
            }
        );

        if (id > 0) {
            edit_url = "/directoryservice/" + "idmap_" + idmap_backend + "/" + id + "/";
        }
...
```
Here we allow `xhr` API handle the request and do the JSON processing. And, in case invalid JSON is returned an error handler is executed, which simply sets `id=0;` as an indication of the non-existing backend. That in turn returns empty `edit_url`, which we handle in `directoryservice_idmap_onclick ()`.

We possibly can return `edit_url` directly from the request handler, but I'm not certain, that it's safe.

And this code works, I swear :)